runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1867,6 +1873,28 @@ private void sendWorkerUpdatesToDataflowService(
           cumulativeCounters.extractUpdates(false, DataflowCounterUpdateExtractor.INSTANCE));
       counterUpdates.addAll(
           deltaCounters.extractModifiedDeltaUpdates(DataflowCounterUpdateExtractor.INSTANCE));
+      if (hasExperiment(options, "beam_fn_api")) {
+        while (!this.pendingMonitoringInfos.isEmpty()) {
+          final CounterUpdate item = this.pendingMonitoringInfos.poll();
```

Comment:
The SDK harness will give you the full value of the counter on each MonitoringInfo. Don't you need to convert it to a delta, since the last reported value? Or does setting cumulative have DFE solve this internally?

You might want to confirm with @edre.

Reply:
It's ok to just report the full value as the delta, but only as long as we only report counters when finishing the work item.

Reply:
Non-cumulative means delta.